src/lib/array.jl

```diff
@@ -550,8 +550,7 @@ end
 
 @inline function ∇broadcast(f::F, args::Vararg{Any,N}) where {F,N}
   y = broadcast(f, data.(args)...)
-  eltype(y) <: Real || return y
-  eltype(y) == Bool && return y
+  (eltype(y) <: Real && eltype(y) !== Bool) || return y
```

Comment:
Did this line need to change for stability, or is it just line 575? It's not immediately obvious to me whether this change is needed to fix inference.

Reply:
No, it's not needed to fix the inference IIRC. While debugging the issue mentioned above I noticed the check `eltype(y) == Bool` in the call stack and changed it to `===` (or rather `!==` in this implementation) to rule out any inference problems there. If desired, I can change it back to the old implementation - I just assumed that `===` (or `!==`) would generally be superior in this case so I kept it.

Reply:
Nope -- I'm perfectly happy with this change, just wanted to make sure that I had not misunderstood.